ZohoCRMCharts/Classes/ZCRMComparator.swift

```diff
@@ -0,0 +1,637 @@
+		didSet {
+			if negativeColor != nil {
+				self.renderOptions.negativeColor = negativeColor
+			}
+		}
+	}
+	
+	public var neutralColor: UIColor! {
+		didSet {
+			if neutralColor != nil {
+				self.renderOptions.positiveColor = positiveColor
+			}
+		}
+	}
+	
+	public var classicHeaderRowColor: UIColor! {
+		didSet {
+			if classicHeaderRowColor != nil {
+				self.renderOptions.classicHeaderRowColor = classicHeaderRowColor
+			}
+		}
+	}
+	
+	public var elegantDiffColor: UIColor! {
+		didSet {
+			if elegantDiffColor != nil {
+				self.renderOptions.elegantDiffColor = elegantDiffColor
+				self.updateHeaders()
+			}
+		}
+	}
+	
+	public var columnWidth: CGFloat!
+	public let title: String
+	public let type: ZCRMCharts.ZCRMComparatorType
+	public let groupings: ZCRMComparatorGroupings
+	public let chunks: [ZCRMComparatorChunk]
+	fileprivate let titleView: UILabel = UILabel()
+	fileprivate var renderOptions: ComparatorRenderOptions = ComparatorRenderOptions()
+	fileprivate var collectionView: UICollectionView!
+	fileprivate let collectionViewFlowLayout: UICollectionViewFlowLayout = UICollectionViewFlowLayout()
+	fileprivate let scrollView: UIScrollView = UIScrollView()
+	fileprivate let chunksView: UIStackView = UIStackView()
+	fileprivate let headersView: UIStackView = UIStackView()
+	fileprivate var chunkDatas: [ZCRMChunkData] = []
+	fileprivate static var groupingsDefaultImage: UIImage!
+	
+	internal var viewConstraints: [NSLayoutConstraint] = []
+	
+	public init(title: String, type: ZCRMCharts.ZCRMComparatorType, groupings: ZCRMComparatorGroupings, chunks: [ZCRMComparatorChunk]) {
+		self.title = title
+		self.type = type
+		self.groupings = groupings
+		self.chunks = chunks
+		super.init(frame: .zero)
+		self.render()
+	}
+	
+	public init(frame: CGRect, title: String, type: ZCRMCharts.ZCRMComparatorType, groupings: ZCRMComparatorGroupings, chunks: [ZCRMComparatorChunk]) {
+		self.title = title
+		self.type = type
+		self.groupings = groupings
+		self.chunks = chunks
+		super.init(frame: frame)
+		self.render()
+	}
+	
+	required public init?(coder aDecoder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+	
+	public override func layoutSubviews() {
+		self.deactivateConstraints()
+		self.addConstraints()
+	}
+	
+	public func reloadData()  {
+		
+		if self.dataSource != nil {
+			self.setChunkDatas()
+		}
+	}
+	
+	public func reloadGroupImages() {
+		
+		if self.type != .classic && self.dataSource != nil {
+			self.setGroupImages()
+		}
+	}
+}
+
+fileprivate extension ZCRMComparator {
+	
+	fileprivate var columCount: Int {
+		return self.type == .classic ? self.chunks.count : self.groupings.groups.count
+	}
+	
+	fileprivate var rowCount: Int {
+		return self.type == .classic ? self.groupings.groups.count + 1 : self.chunks.count
+	}
+	
+	fileprivate var cellWidth: CGFloat {
+		
+		if self.columnWidth != nil && self.type != .sport {
+			return columnWidth
+		}
+		var cellWidth: CGFloat = 150
```

Comment:
Don't use a constant value. Cell width depends on view width.

Reply:
For the table, we need to have a constant. If it depends on the width of the parent there will no scrolling and no difference between the landscape and portrait mode.

Reply:
Constant should depend on the view width or height(Eg: 20/30 percentage)